rfcs/0088-commit-message-guideline.md

```diff
@@ -0,0 +1,181 @@
+  when generating the release notes (enforced);
+* Use an imperative mood. Avoid titles like “Added operator X” and “Updated
+  image Y in the CI”, instead use the forms “Add feature X” and “Update image Y
+  in the CI” instead;
+* Observe proper use of caps at the beginning (uppercase for the first letter)
+  and for acronyms, like, for instance, TVM, FVP, OpenCL. Hence instead of
+  “fix tvm use of opencl library”, write it as “Fix TVM use of OpenCL library”;
+* No period at the end of the title is necessary.
+
+_PR/commit body_:
+
+* Guarantee a body exists (enforced);
+* Don’t use Github usernames in body text, like @username (enforced);
+* Avoid “bullet” commit message bodies: “bullet” commit message bodies are not
+  bad per se, but “bullet” commit messages without any description or
+  explanation is likely as bad as commits without any description, rationale,
+  or explanation in the body.
+
+For minor deviations from these guidelines, the community will normally favor
+reminding the contributor of this policy over reverting or blocking a commmit /
+PR.
+
+Commits and PRs without a title and/or a body are not considered minor
+deviations from these guidelines and hence must be avoided.
+
+Most importantly, the contents of the commit message, especially the body,
+should be written to convey the intention of the change, so it should avoid
+being vague. For example, commits with a title like “Fix”, “Cleanup”, and
+“Fix flaky test” and without any body text should be avoided. Also, for the
+review, it will leave the reviewer wondering about what exactly was fixed or
+changed and why the change is necessary, slowing the review.
+
+Below is an example that can be used as a model:
+
+> [microTVM] Zephyr: Remove zephyr_board option from build, flash, and open_transport methods
+>
+> Currently it’s necessary to pass the board type via ‘zephyr_board’ option to
+> the Project API build, flash, and open_transport methods.
+>
+> However, since the board type is already configured when the project is
+> created (i.e. when the generate_project method is called), it’s possible to
+> avoid this redundancy by obtaining the board type from the project
+> configuration files.
+>
+> This commit adds code to obtain the board type from the project CMake files,
+> removing this option from build, flash, and open_transport methods, so it’s
+> only necessary to specify the ‘zephyr_board’ option when calling
+> generate_project.
+>
+> This commit also moves the ‘verbose’ and ‘west_cmd’ options from ‘build’
+> method to ‘generate_project’, reducing further the number of required options
+> when building a project, since the ‘build’ method is usually called more often
+> than the ‘generate_project’.
+
+If a PR is created from more than one commit (a patchset), then ideally the
+changes should be split into commits in a reasonable way. For instance, if it’s
+a fix and so it also adds a regression test, it’s a good practice to split the
```

Comment:
It might be good to allow a bit of flexibility here. As testcases together with changes also have its own set of benefits (in reducing the overall overhead of crafting a PR) and allow us to focus on the most important part -- the message itself being descriptive

Reply:
I think we might even want the exact opposite guidance here - a fix should always come with a test case otherwise we don't know that anything's been fixed.

Reply:
@tqchen @mbaret I think I'm agree with both of you here, but clearly I failed to express it in the text :) Let me try to explain this paragraph better so you can tell me if that's indeed the case.

I think we are all agreeing that a fix should always come with a testcase.

I'm not saying here that the fix should be submitted in a PR and the testcase be submitted in another PR. I'm departing from the fact that we squash the commits of a PR anyways in the end (unfortunately, I have to say), so I'm just trying to say that for sake of reviewing it's recommended (or desirable) the PR (which has two commits: fix + testcase in the example) must have these two commits split in a reasonable way, with due commit messages, like not as single commit with fix code and testcase code intertwined all together -- as sometimes it happens. I'm trying here to just give some advice to actually ease the review process.

But OTOH I agree with @tqchen about the focus on the most important part, the message itself being descriptive. I'm ok on removing this paragraph. @leandron do you agree?

Reply:
Yes, I agree we can remove the paragraph.

Reply:
@tqchen can you check and mark this as resolved if applicable?